doc/rationale.md

```diff
@@ -0,0 +1,31 @@
+# Ignition rationale
+
+Ignition is a distribution-agnostic, provisioning utility. When a machine boots for the first time, Ignition will manipulate it in various ways to set up the machine for operations.
+
+There are several principles that have guided Ignition's design. This document aims to explain these principles and how they impact Ignition's design.
+
+## Ignition only runs on the first boot
+
+Ignition is a provisioning tool; not a configuration management tool. Ignition encourages immutable infrastructure, wherein if a user wishes to modify a machine, they should discard the old node and re-provision the machine. This keeps the user's machines in a well known state with relatively simple tooling.
+
+To this end, Ignition runs on the first boot of a machine. Being a general provisioning utility however, there's nothing stopping a user from using Ignition to set up some configuration management tool. This may even make a lot of sense depending on the organization and requirements the user is trying to meet.
+
+It's worth noting that from the user's perspective this first boot is not special in any way. The system is fully provisioned and ready for use from the moment the user can interact with it.
```

Comment:
Drop "It's worth noting".

s/from the moment the user can interact with it/once fully booted

(don't encourage users interacting with machines)